statements/src/test/java/org/lukawska/trainsmart/statements/presentation/exception/GlobalExceptionHandlerTest.java

```diff
@@ -0,0 +1,112 @@
+package org.lukawska.trainsmart.statements.presentation.exception;
+
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.ConstraintViolationException;
+import org.junit.jupiter.api.Test;
+import org.lukawska.trainsmart.statements.application.exception.ExceptionType;
+import org.lukawska.trainsmart.statements.application.exception.StatementException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.context.request.WebRequest;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.lukawska.trainsmart.statements.testutil.ExceptionTestData.mockViolation;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class GlobalExceptionHandlerTest {
+
+    private final GlobalExceptionHandler exceptionHandler = new GlobalExceptionHandler();
+
+    @Test
+    void shouldHandleStatementExceptionWhenStatementNotFound() {
+        //given
+        final StatementException statementException = new StatementException(ExceptionType.STATEMENT_NOT_FOUND);
+
+        //when
+        ProblemDetail result = exceptionHandler.handleStatementException(statementException);
+
+        //then
+        ProblemDetailAssert.then(result)
+                           .isNotNull()
+                           .hasStatus(HttpStatus.NOT_FOUND)
+                           .hasDetail(statementException.getMessage())
+                           .hasTitle("Statement exception");
+
+    }
+
+    @Test
+    void shouldHandleGenericException() {
+        //given
+        Exception exception = new Exception(UUID.randomUUID().toString());
+
+        //when
+        ProblemDetail result = exceptionHandler.handleGenericException(exception);
+
+        //then
+        ProblemDetailAssert.then(result)
```

Comment:
to tez moze zostac tak samo bo to troche jak builder. mowie tylko zebys nie musiala tego przerabiac rowniez